google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/ConnectionWorker.java

```diff
@@ -278,6 +303,83 @@ static String getRoutingHeader(String streamName, String location) {
     return project + "locations/" + location;
   }
 
+  private Attributes buildOpenTelemetryAttributes() {
+    AttributesBuilder builder = Attributes.builder().put(telemetryKeyStreamId, this.streamName);
+    builder.put(telemetryKeyWriterId, this.writerId);
+    if ((this.traceId != null) && !this.traceId.isEmpty()) {
+      String[] traceIdParts = this.traceId.split(":", 5);
+      for (int i = 0; i < traceIdParts.length; i++) {
+        builder.put(telemetryKeysTraceId.get(i), traceIdParts[i]);
+      }
+    }
+    return builder.build();
+  }
+
+  @VisibleForTesting
+  Attributes getTelemetryAttributes() {
+    // streamName can change due to multiplexing. If it has changed, update the attributes.
+    String originalStreamNameAttribute = telemetryAttributes.get(telemetryKeyStreamId);
+    if ((originalStreamNameAttribute != null)
+        && !originalStreamNameAttribute.equals(this.streamName)) {
+      AttributesBuilder builder = telemetryAttributes.toBuilder();
+      builder.put("streamId", this.streamName);
+      telemetryAttributes = builder.build();
+    }
+    return telemetryAttributes;
+  }
+
+  private void registerOpenTelemetryMetrics() {
+    MeterProvider meterProvider = Singletons.getOpenTelemetry().getMeterProvider();
+    writeMeter =
+        meterProvider
+            .meterBuilder("com.google.cloud.bigquery.storage.v1.write")
```

Comment:
@shollyman is this namespace going to be shared by all languages?

Reply:
I'm also curious to know this. I don't see where the Go client lib defines the namespace for the metrics.

Reply:
The com.* prefix is a java packaging idiom.  The equivalent for go would likely be something like "cloud.google.com/go/bigquery/storage/managedwriter". That said, I'm not familiar enough here. It seems like Go exposes a "Resource" abstraction for defining the naming prefix.